python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -543,18 +543,37 @@ MlirModule synthesizeKernel(const std::string &name, MlirModule module,
   auto isLocalSimulator = platform.is_simulator() && !platform.is_emulated();
   auto isSimulator = isLocalSimulator || isRemoteSimulator;
 
-  cudaq::opt::ArgumentConverter argCon(name, unwrap(module), isSimulator);
+  cudaq::opt::ArgumentConverter argCon(name, unwrap(module));
   argCon.gen(runtimeArgs.getArgs());
-  std::string kernName = cudaq::runtime::cudaqGenPrefixName + name;
-  SmallVector<StringRef> kernels = {kernName};
-  std::string substBuff;
-  llvm::raw_string_ostream ss(substBuff);
-  ss << argCon.getSubstitutionModule();
-  SmallVector<StringRef> substs = {substBuff};
+
+  // Store kernel and substitution strings on the stack.
+  // We pass string references to the `createArgumentSynthesisPass`.
+  mlir::SmallVector<std::string> kernels;
+  mlir::SmallVector<std::string> substs;
+  for (auto &kInfo : argCon.getKernelSubstitutions()) {
+    {
+      std::string kernName =
+          cudaq::runtime::cudaqGenPrefixName + kInfo.getKernelName().str();
+      kernels.emplace_back(kernName);
+    }
+    {
+      std::string substBuff;
+      llvm::raw_string_ostream ss(substBuff);
+      ss << kInfo.getSubstitutionModule();
+      substs.emplace_back(substBuff);
+    }
```

Comment:
```suggestion
  
```
The extra braces aren't needed and are distracting. clang is smart enough to be able to determine object lifetimes and optimize dtor calls, etc. I think we can just let the compiler do what it does best here.